pkg/enqueue-bundle/DependencyInjection/Configuration.php

```diff
@@ -50,57 +50,46 @@ public function getConfigTreeBuilder(): TreeBuilder
 
         $transportFactory = new TransportFactory('default');
 
-        /** @var ArrayNodeDefinition $transportNode */
-        $transportNode = $rootNode->children()->arrayNode('transport');
-        $transportNode
-            ->beforeNormalization()
-            ->always(function ($value) {
-                if (empty($value)) {
-                    return ['default' => ['dsn' => 'null:']];
-                }
-                if (is_string($value)) {
-                    return ['default' => ['dsn' => $value]];
-                }
-
-                if (is_array($value) && array_key_exists('dsn', $value)) {
-                    return ['default' => $value];
-                }
-
-                return $value;
-            });
-        $transportPrototypeNode = $transportNode
-            ->requiresAtLeastOneElement()
-            ->useAttributeAsKey('key')
-            ->prototype('array')
-        ;
+        $transportConfig = $transportFactory->getConfiguration('transport');
+        $transportConfig->isRequired();
 
-        $transportFactory->addTransportConfiguration($transportPrototypeNode);
+        $consumerConfig = $transportFactory->getQueueConsumerConfiguration('consumption');
 
-        $consumptionNode = $rootNode->children()->arrayNode('consumption');
-        $transportFactory->addQueueConsumerConfiguration($consumptionNode);
+        $clientConfig = (new ClientFactory('default'))->getConfiguration('client', $this->debug);
```

Comment:
What if make this getConfiguration methods static? It  is confusing to create a default factory and get a configuration from it. 

@ASKozienko  WDYT?